config/routes.rb

```diff
@@ -1,4 +1,8 @@
 Rails.application.routes.draw do
+  get '/', to: 'users#index', as: :users
+  get '/users/:id', to: 'users#show', as: :user
+  get '/users/:id/posts', to: 'posts#index', as: :user_posts
+  get '/users/:user_id/posts/:id', to: 'posts#show', as: :user_post
```

Comment:
- [OPTIONAL] While this is going to work fine, I will encourage you to take advantage of resources by having something like:
```
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```

With this, you have shorter code to write and you keep it simple, stupid.